Source/FakeItEasy/Core/FakeWrapperConfigurator.cs

```diff
@@ -47,6 +47,13 @@ public IFakeOptionsBuilder<T> WithAdditionalAttributes(IEnumerable<CustomAttribu
 
         public IFakeOptionsBuilder<T> Implements(Type interfaceType)
         {
+            Guard.AgainstNull(interfaceType, "interfaceType");
+
+            if (!interfaceType.IsInterface)
+            {
+                throw new ArgumentException("The specified type is not an interface");
+            }
+

```

Comment:
Yeah, it's not obvious at first glance, but since both classes are internal (or better) and built the way they are to support the fluent syntax, I think it's safe to collapse the checks onto `FakeOptions.AddInterfaceToImplement`, if you don't mind. All else being equal, I always vote for fewer lines of code. Unfortunately at this point, that's more work for you. Then again, if you update the exception message with the supplied type, you'll only have to do it once. :grinning: